add missing requirements for internal file pointers (#1134)
php-fig/fig-standards: accepted/PSR-17-http-factory.md

}
```

> The created instance MUST return an empty body-stream in read and write mode.

Reviewer: As we discussed elsewhere, the spec itself cannot be changed. We can only update the meta document with errata.

Reviewer: At least as far as adding **MUST** statements go.

Perhaps it is acceptable to add **SHOULD** statements... @php-fig/secretaries ?

Author: Because the bylaws say so?

I think you should vote on an exemption here - the spec is considerably less useful without these amendments being visible where they should be. If an implementer has to dig portions of the spec out of the meta section (which isn't even in the same file) that's quite a serious problem.

If what you're saying is, this isn't an "amendment" and can't be considered "errata", then it sounds like we're back to the idea of deprecating PSR-17.

Given the fact that the current spec does *not* guarantee interoperable implementations, in my opinion, this *is* errata. The definition of errata in the bylaws is somewhat ambiguous.

As I've argued before, this isn't a "breaking change" in the sense that it's going back on something that was already specified - we aren't making changes versus the previous specification, we're adding something that was unspecified and is currently inconsistent among existing implementations.

I sincerely hope we can amend rather than putting the entire community through the painful process of replacing the entire standard with a new one?

     * Create a new stream from a string.
     *
     * The stream SHOULD be created with a temporary resource.
     * 
     * The stream MUST be created with the current position of the file read/write

>      * pointer at the beginning of the stream. (Errata 9.1)

Reviewer: @mindplay-dk on closer inspection, this one is not correct. Given that this is an "open + write" operation, it would be natural for the pointer to be at the end of the stream and there is no guarantee that the stream can be rewound, making it impossible to place the pointer at the beginning of the stream.

Instead, I believe this can be combined with the previous line and read:

> The stream SHOULD be created with a temporary resource, and the file read/write pointer SHOULD be placed consistently with an `fwrite()` operation.

>      * The stream SHOULD be created with a temporary resource.
>      * 
>      * The stream MUST be created with the current position of the file read/write
>      * pointer at the beginning of the stream. (Errata 9.1)

Reviewer: ```suggestion
     * The stream SHOULD be created with a temporary resource, and the file pointer SHOULD
     * be placed consistently with an fwrite() operation. (Errata 9.1)
```

     *
     * The `$filename` MAY be any string supported by `fopen()`.
     *
     * The stream MUST be created with the current position of the file read/write
     * pointer positioned consistently with that of `fopen()` with regards to

>      * the `$mode` flag. (Errata 9.2)

Reviewer: `s/MUST/SHOULD/`

     *
     * The stream MUST be readable and may be writable.
     *

>      * The current read/write position of the given PHP resource MUST NOT be modified. (Errata 9.3)

Reviewer: `s/MUST/SHOULD/`
